src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.tsx

```diff
@@ -177,32 +183,66 @@ const ProteinsListItemInfo = (props: Props) => {
           </>
         )}
 
-        {(proteinSummaryLoadingState === LoadingState.NOT_REQUESTED ||
-          !product) && (
-          <div className={styles.statusContainer}>
-            <CircleLoader />
-          </div>
-        )}
-
-        {proteinSummaryLoadingState === LoadingState.ERROR && product && (
-          <div className={styles.statusContainer}>
-            <span className={styles.errorMessage}>Failed to get data</span>
-            <PrimaryButton
-              onClick={() =>
-                setProteinSummaryLoadingState(LoadingState.NOT_REQUESTED)
-              }
-            >
-              Try again
-            </PrimaryButton>
-          </div>
-        )}
+        <StatusContent
+          summaryLoadingState={summaryLoadingState}
+          domainsLoadingState={domainsLoadingState}
+          setSummaryLoadingState={setSummaryLoadingState}
+          setDomainsLoadingState={setDomainsLoadingState}
+        />
 
         <div className={styles.keyline}></div>
       </div>
     </div>
   );
 };
 
+type StatusContentProps = {
+  summaryLoadingState: LoadingState;
+  domainsLoadingState: LoadingState;
+  setSummaryLoadingState: (loadingState: LoadingState) => void;
+  setDomainsLoadingState: (loadingState: LoadingState) => void;
+};
+
+const StatusContent = (props: StatusContentProps) => {
+  if (
+    props.domainsLoadingState === LoadingState.LOADING ||
+    props.summaryLoadingState === LoadingState.LOADING
+  ) {
+    return (
+      <div className={styles.statusContainer}>
+        <CircleLoader />
+      </div>
+    );
+  }
+
+  let retryHandler;
+
+  if (
+    props.domainsLoadingState === LoadingState.ERROR &&
+    props.summaryLoadingState === LoadingState.ERROR
+  ) {
+    retryHandler = () => {
+      props.setSummaryLoadingState(LoadingState.LOADING);
+      props.setDomainsLoadingState(LoadingState.LOADING);
+    };
+  } else if (props.domainsLoadingState === LoadingState.ERROR) {
+    retryHandler = () => {
+      props.setDomainsLoadingState(LoadingState.LOADING);
+    };
+  } else if (props.summaryLoadingState === LoadingState.ERROR) {
+    retryHandler = () => {
+      props.setSummaryLoadingState(LoadingState.LOADING);
+    };
+  }
```

Comment:
this can probably be written shorter and easier:

```
const retryHandler = () => {
  if (props.domainsLoadingState === LoadingState.ERROR) {
    props.setDomainsLoadingState(LoadingState.LOADING);    
  }
  if (props.summaryLoadingState === LoadingState.ERROR) {
    props.setSummaryLoadingState(LoadingState.LOADING);
  }
};
```

Reply:
That's done